common/presenters/single-approved-requests-to-table-component.js

```diff
@@ -0,0 +1,84 @@
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+const componentService = require('../services/component')
+
+const moveToCardComponent = require('./move-to-card-component')
+const tablePresenters = require('./table')
+
+function singleRequestsToTable(moves) {
+  const tableConfig = [
+    {
+      head: {
+        text: i18n.t('name'),
+        attributes: {
+          width: '220',
+        },
+      },
+      row: {
+        html: data => {
+          const card = moveToCardComponent({
+            isCompact: true,
+            hrefSuffix: '/review',
+          })(data)
+          return componentService.getComponent('appCard', card)
+        },
+        attributes: {
+          scope: 'row',
+        },
+      },
+    },
+    {
+      head: {
+        text: i18n.t('collections::labels.created_at'),
+        attributes: {
+          width: '120',
+        },
+      },
+      row: {
+        text: data => filters.formatDate(data.created_at),
+      },
+    },
+    {
+      head: {
+        text: i18n.t('collections::labels.from_location'),
+      },
+      row: {
+        text: 'from_location.title',
+      },
+    },
+    {
+      head: {
+        text: i18n.t('collections::labels.to_location'),
+      },
+      row: {
+        text: 'to_location.title',
+      },
+    },
+    {
+      head: {
```

Comment:
Do we really need this additional module?

Could we not pass an additional arg to singleRequestsToTable, eg. date_type

```
head: {
  text: i18n.t(`collections::labels.${date_type === 'actual' ? '' : 'earliest_' }move_date`),
  attributes: {
    width: '120',
  },
},
row: {
  text: data => filters.formatDate(data[date_type === 'actual' ? 'date' : 'date_from']),
},
```

Reply:
I think I'm leaning towards what @solidgoldpig said in that this feels like a lot of extra code to change the presentation of one column.

In essence it's still about transforming a single request into a table format but it sounds like the desired business rules are that if an actual date has been defined, we want to display it instead of the proposed dates.

I'd probably consider updating the existing table with something like this:

```javascript
{
  head: {
    text: i18n.t(data.date ? 'collections::labels.move_date' : 'collections::labels.earliest_move_date'),
    attributes: {
      width: '120',
    },
  },
  row: {
    text: data => filters.formatDate(data.date || data.date_from),
  },
},
```

Deep down, the logic is probably better being determined by the presence of the `date` attribute rather than the status of the move. A move can't be "approved" without a date being sent as part of the approval.

Reply:
yep, @teneightfive's suggestion to use the attribute itself to determine the column is a better way of handling it.